packages/lib/src/spatial-navigation/components/FocusableView.tsx

```diff
@@ -1,21 +1,53 @@
 import { SpatialNavigationNode, SpatialNavigationNodeDefaultProps } from './Node';
-import { View, ViewStyle } from 'react-native';
-import { forwardRef } from 'react';
+import { Platform, View, ViewStyle } from 'react-native';
+import { forwardRef, useImperativeHandle, useRef } from 'react';
 import { SpatialNavigationNodeRef } from '../types/SpatialNavigationNodeRef';
+import { useDevice } from '../context/DeviceContext';
 
 type Props = SpatialNavigationNodeDefaultProps & {
   style?: ViewStyle;
   children:
     | React.ReactElement
     | ((props: { isFocused: boolean; isActive: boolean }) => React.ReactElement);
+  onMouseEnterProps?: () => void;
+  onMouseLeaveProps?: () => void;
 };
 
 export const SpatialNavigationFocusableView = forwardRef<SpatialNavigationNodeRef, Props>(
-  ({ children, style, ...props }: Props, ref) => {
+  ({ children, style, onMouseEnterProps, onMouseLeaveProps, ...props }: Props, ref) => {
+    const { deviceType } = useDevice();
+    const nodeRef = useRef<SpatialNavigationNodeRef>(null);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        focus: () => nodeRef.current?.focus(),
+      }),
+      [nodeRef],
+    );
+
+    const webProps = Platform.select({
+      web: {
+        onMouseEnter: () => {
+          if (onMouseEnterProps) {
+            onMouseEnterProps();
```

Comment:
I wonder if we should still focus even if onMouseEnter has been provided?

Reply:
We considered that if someone provides onMouseEnter it means that it wants to bypass the normal behavior, but that's totally a choice we can challenge !

Reply:
Maybe we should ignore the focus only if onMouseEnterProps returns true? Same as the event propagation mechanism?

Reply:
Done :D